code/datums/Emote_system/emote.dm

```diff
@@ -0,0 +1,574 @@
+		return 1
+	if(!isliving(user))
+		return
+	var/mob/living/L = user
+	if(L.silent)
+		return 1
+
+/datum/emote/proc/checkForParams(var/list/params)
+	for(var/p in params)
+		if(p == "target")
+			continue
+		return 1
+
+/datum/emote/proc/addExtras(var/mob/user, var/list/params, var/message = "")
+	if(!message)
+		return
+	if(startText)
+		message = "[startText] [message]"
+	message = "<span class='[emoteSpanClass]'>[message]</span>"
+	return message
+
+/datum/emote/proc/standardMessage(var/mob/user, var/list/params)
+	var/message = "<span class='[userSpanClass]'>\The [user]</span> [text]"
+	if("target" in params)
+		message = addTarget(user, params, message)
+	return message
+
+/datum/emote/proc/mimeMessage(var/mob/user, var/list/params)
+	if(!mimeText)
+		return
+	if(checkForParams(params))
+		return paramMimeMessage(user, params)
+	var/message = "<span class='[userSpanClass]'>\The [user]</span> [mimeText]"
+	if(message && "target" in params)
+		message = addTarget(user, params, message)
+	return message
+
+/datum/emote/proc/muzzleMessage(var/mob/user, var/list/params)
+	var/message = "<span class='[userSpanClass]'>\The [user]</span> makes a "
+	if(muzzledNoise)
+		message += "[muzzledNoise] "
+	message += "noise"
+	return message
+
+// if the emote takes a non target parameter, set up  and return the with parameter version in here
+/datum/emote/proc/paramMessage(var/mob/user, var/list/params)
+	return
+
+// as above, but for mimes when there is mimeText
+/datum/emote/proc/paramMimeMessage(var/mob/user, var/list/params)
+	return
+
+/datum/emote/proc/addTarget(var/mob/user, var/list/params, var/message = "")
+	if(!canTarget)
+		return message
+	if(!params["target"])
+		return message
+	if(params["target"] == user)
+		message += " [targetText][getHimself(user)]"
+		return message
+	message += " [targetText]\the [params["target"]]"
+	return message
+
+
+// What you should see when you perform the emote
+/datum/emote/proc/createSelfMessage(var/mob/user, var/list/params, var/message = "")
+	if(!selfText)
+		return message
+
+	message = replacetext(message, text, selfText)
+	message = changeTextMacros(user, message)
+
+	if(mimeSelf)
+		message = replacetext(message, mimeText, mimeSelf)
+
+	if(startText && !selfStart)
+		var/start = findtextEx(message, startText)
+		var/end = start + lentext(startText) + 1
+		message = copytext(message, 1, start) + copytext(message, end, lentext(message) + 1)
+
+	message = replaceMobWithYou(user, message)
+
+	return message
+
+/datum/emote/proc/replaceMobWithYou(var/mob/M, var/message = "", var/mob/user)	// user passed in in case an override needs to do something different if M != user (see johnny) VB
+
+	message = replacetext(message, "\The [M]", "You")
+	message = replacetext(message, "\the [M]", "you")
+
+	return message
+
+/datum/emote/proc/processMessage(var/mob/user, var/list/params, var/message = "")
+	var/visualOrAudible = audible + 1
+	testing("message = [message]")
```

Comment:
yeah, It'll be gone in a few minutes, just doing last tests on the adding of the . if there's no ending punctuation, will be pushing final version very soon